### DISCUSSION TITLE
How to mount secret files

### DISCUSSION THREAD

**Discussion by villesau** [vercel/vercel#7628] [Help] [answered]
Hi,

How do I mount secret files that cannot reside in git? Good example is to set ssl certificates for Prisma: https://www.prisma.io/docs/concepts/database-connectors/postgresql

There is no way to pull those from env variables, but instead they need to be as a file.

Not having this prevents from using e.g google cloud sql and likely many other databases too.

**Comment by liftyourgame** [comment]
If the file is short try putting the content in an environment variable and then write it to disk just before you need it.

**Comment by leerob** [comment, marked as answer]
You could encrypt the contents, allowing you to store in git and decrypt at runtime: https://vercel.com/support/articles/how-do-i-workaround-vercel-s-4-kb-environment-variables-limit

**Reply by villesau** [reply]
That's actually the most feasible sounding workaround so far!  Though, the endpoint based solution linked sounds terrible. There's actually EJSON format built by Shopify for storing secrets in git: https://github.com/Shopify/ejson I'm not sure if there's good JS bindings for it though. @leerob Maybe Vercel could build something closer to first class around EJSON?

**Reply by vivekseth** [reply]
FWIW this link is now showing this error


<img width="1425" alt="Screenshot 2023-12-28 at 3 14 03 PM" src="https://github.com/vercel/vercel/assets/3159713/989e9ecf-a996-47fb-8215-2fe9fe388538">

**Comment by brian-dlee** [comment]
After fighting with an elegant way to do this for some time, I decided to create a solution. I'm just making the rounds on these related discussions to help others find it. This tool helps get large environment variables encrypted and "smuggled" into a Vercel deployment and includes some functions to easily read the contents at runtime.

https://github.com/brian-dlee/smuggler
